src/AspNetCoreModuleV2/OutOfProcessRequestHandler/outofprocess/outprocessapplication.cpp

```diff
@@ -47,6 +47,16 @@ OUT_OF_PROCESS_APPLICATION::Initialize(
         }
     }
 
+    if (m_pFileWatcherEntry == NULL)
+    {
+        m_pFileWatcherEntry = new  FILE_WATCHER_ENTRY(g_pFileWatcher);
```

Comment:
Shouldn't the file watcher entry be created in the application?

Reply:
I just want to make the application.h simple and abstract enough. an application may not need do file watching for appoffline.

Reply:
Add `FILE_WATCHING_APPLICATION` then